akka-actor/src/main/scala/akka/dispatch/Future.scala

```diff
@@ -80,13 +81,12 @@ object ExecutionContexts {
    * This is an execution context which runs everything on the calling thread.
    * It is very useful for actions which are known to be non-blocking and
    * non-throwing in order to save a round-trip to the thread pool.
+   *
+   * INTERNAL API
    */
-  private[akka] object sameThreadExecutionContext extends ExecutionContext with BatchingExecutor {
```

Comment:
can we also deprecated BatchingExecutor now?

Reply:
I've separated out all the BatchingExecutor stuff from this PR because I think these changes are isolated, I think we still need of `BatchingExecutor` for now.